subgraphs/erc20/schema.graphql

```diff
@@ -0,0 +1,293 @@
+  " Smart contract address of the token "
+  id: ID!
+
+  " Name of the token, mirrored from the smart contract "
+  name: String!
+
+  " Symbol of the token, mirrored from the smart contract "
+  symbol: String!
+
+  " The number of decimal places this token uses, default to 18 "
+  decimals: Int!
+
+  " If token transfers are paused, derived from paused event "
+  isPaused: Boolean
+
+  " Token Holders Balance "
+  holdersBalance: [AccountBalance!]! @derivedFrom(field: "token")
+
+  " Number of token holders "
+  holderCount: BigInt!
+
+  " Total number of token transfer transactions "
+  transferCount: BigInt!
+
+  " Total number of token mint transactions "
+  mintCount: BigInt!
+
+  " Total number of token burn transactions "
+  burnCount: BigInt!
+
+  " Total token supply "
+  totalSupply: BigDecimal!
+
+  " Total token burned "
+  totalBurned: BigDecimal!
+
+  " Total token minted "
+  totalMinted: BigDecimal!
+
+  " List of token transfers "
+  transfers: [Transfer!]! @derivedFrom(field: "token")
+
+  ##### Snapshots #####
+
+  " Daily usage metrics for this Token "
+  dailyUsageMetrics: [UsageMetricsDailySnapshot!]! @derivedFrom(field: "token")
+
+  " Hourly usage metrics for this Token "
+  hourlyUsageMetrics: [UsageMetricsHourlySnapshot!]!
+    @derivedFrom(field: "token")
+
+  dailyTokenSnapshot: [TokenDailySnapshot!]! @derivedFrom(field: "token")
+}
+
+###############################
+####### Token Timeseries ######
+###############################
+
+type UsageMetricsDailySnapshot @entity {
+  " { Token Address } - {# of days since Unix epoch time} "
+  id: ID!
+
+  " Token this snapshot is associated with "
+  token: Token!
+
+  " # of accounts holding this token "
+  dailyHolderCount: Int!
+
+  " # of unique daily active users "
+  dailyActiveUsers: Int!
+
+  " # of cumulative unique users "
+  cumulativeUniqueUsers: Int!
+
+  " Total number of transactions occurred in a day. Transactions include all entities that implement the Event interface. "
+  dailyTransactionCount: Int!
+
+  " Total number of mints in a day "
+  dailyMintCount: Int!
+
+  " Total number of token minted in a day "
+  dailyMintAmount: BigInt!
+
+  " Total number of burns in a day "
+  dailyBurnCount: Int!
+
+  " Total number of token burnt in a day "
+  dailyBurnAmount: BigInt!
+
+  " Block number of this snapshot "
+  blockNumber: BigInt!
+
+  " Timestamp of this snapshot "
+  timestamp: BigInt!
+}
+
+type UsageMetricsHourlySnapshot @entity {
+  " { Token Address } - {# of hours since Unix epoch time} "
+  id: ID!
+
+  " Token this snapshot is associated with "
+  token: Token!
+
+  " # of accounts holding this token "
+  hourlyHolderCount: Int!
+
+  " # of unique hourly active users "
+  hourlyActiveUsers: Int!
+
+  " # of cumulative unique users "
+  cumulativeUniqueUsers: Int!
+
+  " Total number of transactions occurred in an hour. Transactions include all entities that implement the Event interface. "
+  hourlyTransactionCount: Int!

```

Comment:
since tokens have very basic events (and don't have calls to handle), maybe we should just rename all transactions to events?